lua/starfall/libs_sv/prop.lua

```diff
@@ -279,15 +278,46 @@ function props_library.createComponent(pos, ang, class, model, frozen)
 	return ewrap(comp)
 end
 
+--- Get a list of all spawnable sents.
+-- @param ordered True to get an ordered list
+-- @return The table
+function props_library.getSpawnableSents(ordered)
+	local tbl = {}
+	
+	local add
+	if ordered then
+		add = function(list_name)
+			tbl[list_name] = {}
+			for class, _ in pairs(list.GetForEdit(list_name)) do
+				table.insert(tbl[list_name], class)
+			end
+		end
+	else
+		add = function(list_name)
+			for class, _ in pairs(list.GetForEdit(list_name)) do
+				table.insert(tbl, class)
+			end
+		end
+	end
+	
+	add("Weapon")
+	add("SpawnableEntities")
+	add("NPC")
+	add("Vehicles")
+	add("starfall_creatable_sent")
+	
+	return tbl
+end
+
 --- Creates a sent.
 -- @param pos Position of created sent
 -- @param ang Angle of created sent
 -- @param class Class of created sent
 -- @param frozen True to spawn frozen
+-- @param data Optional table, additional entity data to be supplied
```

Comment:
I'd specify the data is for specific sents and direct the user to the documentation of those sents